cmd/token.go

```diff
@@ -135,14 +110,41 @@ The OIDC token includes claims regarding the GCE VM, which is verified by Attest
 		}
 
 		key = "gceAK"
-		attestAgent := agent.CreateAttestationAgent(rwc, attestationKeys[key][keyAlgo], verifierClient, principalFetcher, nil, spec.LaunchSpec{}, nil, cloudLogger)
 
 		fmt.Fprintf(debugOutput(), "Fetching attestation verifier OIDC token\n")
-		token, err := attestAgent.Attest(ctx, agent.AttestAgentOpts{Aud: audience, TokenType: "OIDC"})
+
+		challenge, err := verifierClient.CreateChallenge(ctx)
 		if err != nil {
-			return fmt.Errorf("failed to retrieve attestation service token: %v", err)
+			return err
+		}
+
+		principalTokens, err := util.PrincipalFetcher(challenge.Name, mdsClient)
+		if err != nil {
+			return fmt.Errorf("failed to get principal tokens: %w", err)
+		}
+
+		attestation, err := util.GetAttestation(rwc, attestationKeys[key][keyAlgo], challenge.Nonce)
+		if err != nil {
+			return err
 		}
 
+		req := verifier.VerifyAttestationRequest{
+			Challenge:      challenge,
+			GcpCredentials: principalTokens,
+			Attestation:    attestation,
+			TokenOptions:   verifier.TokenOptions{CustomAudience: audience, TokenType: "OIDC"},
+		}
+
+		resp, err := verifierClient.VerifyAttestation(ctx, req)
+		if err != nil {
+			return err
+		}
+		if len(resp.PartialErrs) > 0 {
```

Comment:
`PartialErrs` is acceptable here as this field was originally designed for holding detailed error messages about invalid container signatures for debugging purpose. So instead of directly returning an error, consider logging these errors.